raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -3768,8 +3779,8 @@ public void testHandleLeaderChangeFiresAfterFollowerRegistration(boolean withKip
     public void testObserverFetchWithNoLocalId(boolean withKip853Rpc) throws Exception {
         // When no `localId` is defined, the client will behave as an observer.
         // This is designed for tooling/debugging use cases.
-
-        Set<Integer> voters = Utils.mkSet(1, 2);
+        int localId = randomReplicaId();
```

Comment:
The local replica doesn't have an id. Notice that this test has `new RaftClientTestContext.Builder(OptionalInt.empty(), ...)`.

You can change this line to `int leaderId = randomReplicaId()`, remove the line below `int leaderId = localId` and replace all usage of `localId` with `leaderId`.

Reply:
renamed localId to leaderId and removed the previous leaderId line